### PR TITLE
Prevent injection vulnerability in image page

### LINKED ISSUE #12542: Escape injected URLs in servohtmlparser
In #12539 we almost introduced an XSS vulnerability by injecting the unescaped URL into the neterror.html page. We do the same injection for displaying images. We should find a way to escape these.

### PR DESCRIPTION
**Pull request by johannhof**
This is taking up nox' suggestion from #12542 and creates an img element using Rust code instead of escaping the URL. I will look at the neterror.html URL strings separately, we might do those in a similar way.

To reproduce, visit e.g. the following URL with your vulnerable Servo:

```
https://servo.org/screenshot.png?'onload='document.body.innerHTML=`hacked`'
```

---
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #12542
- [x] These changes do not require tests because this is just fixing up existing behavior and I'm not sure how to test it

r?@jdm

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/12679)

### CODE REVIEW

File: components/script/dom/servohtmlparser.rs [hunk 1]
```diff
@@ -112,9 +117,16 @@ impl AsyncResponseListener for ParserContext {
         match content_type {
             Some(ContentType(Mime(TopLevel::Image, _, _))) => {
                 self.is_synthesized_document = true;
-                let page = format!("<html><body><img src='{}' /></body></html>", self.url);
+                let page = format!("<html><body></body></html>");
```
Comment (notriddle) [inline components/script/dom/servohtmlparser.rs]: Does this really need format still?

Comment (johannhof) [inline components/script/dom/servohtmlparser.rs]: Nope, I just glanced at the block below and that one also has an unneeded `format!`, so I thought I'd stay consistent. I could change both, what would you prefer?

Comment (Manishearth) [inline components/script/dom/servohtmlparser.rs]: Nope 😄

**Comment by Manishearth** [comment]
@bors-servo r-

Please remove the `format!`. While you're at it, try to replace the expect message with something more descriptive.

r=me post nits

**Comment by johannhof** [comment]
@Manishearth updated

**Pull request merged**